lib/combinator/transduce.js

```diff
@@ -0,0 +1,57 @@
+/** @license MIT License (c) copyright 2010-2015 original author or authors */
+/** @author Brian Cavalier */
+/** @author John Hann */
+
+var Stream = require('../Stream');
+
+exports.transduce = transduce;
+
+function transduce(transducer, stream) {
+	return new Stream(new Transduce(transducer, stream.source));
+}
+
+function Transduce(transducer, source) {
+	this.transducer = transducer;
+	this.source = source;
+}
+
+Transduce.prototype.run = function(sink, scheduler) {
+	var xf = this.transducer(new Transformer(sink));
+	return this.source.run(new TransduceSink(xf, sink), scheduler);
+};
+
+function TransduceSink(xf, sink) {
+	this.xf = xf;
+	this.sink = sink;
+}
+
+TransduceSink.prototype.event = function(t, x) {
+	var next = this.xf.step(t, x);
+	if(next != null && next.__transducers_reduced__) {
+		this.sink.end(t, next.value);
+	}
+};
+
+TransduceSink.prototype.end = function(t /*, x */) {
+	this.xf.result(t);
```

Comment:
Great question.  Most.js separates the notion of values "in the stream" and an "end signal value".  This is similar, but not identical, to who ES6 generates subtle separate `yield`ed values (which always have `done === true`) from a final `return`ed value (which has `done === false`).

So, now you've got me thinking that a better approach would be to wrap up `t` and `x` into a `{ time, value }` object and do `this.xf.result({ time: t, value: x })`, and then `Transformer.prototype.result`, could do `this.sink.end(time, value)`.  That would propagate this final value as the "end signal value".